shm/Player.cpp

```diff
@@ -9,8 +9,8 @@ Player::Player(std::unique_ptr<Ship> ship, size_t money) {
 };
 
 size_t Player::countAvailableSpace() {
-    auto occupiedSpace = std::accumulate(begin(ship_->getAllCargos()),
-                                         end(ship_->getAllCargos()),
+    auto occupiedSpace = std::accumulate(begin(getAllCargos()),
+                                         end(getAllCargos()),
                                          0,
                                          [](const Cargo& cargoFirst, const Cargo& cargoSecond) {
                                              return cargoFirst.getAmount() + cargoSecond.getAmount();
```

Comment:
Are you sure this is a proper usage of std::accumulate algorithm? Based on https://en.cppreference.com/w/cpp/algorithm/accumulate the binary operation (lambda in our case) is defined as:

> op - binary operation function object that will be applied. The binary operator takes the current accumulation value a (initialized to init) and the value of the current element b. The signature of the function should be equivalent to the following: `Ret fun(const Type1 &a, const Type2 &b);` 

So it seems that the first parameter of lambda should be current accumulation value. And current value from Cargo vector is the second parameter of lambda.

Reply:
Yeah, my mistake.